src/scripts/main.js

```diff
@@ -1,3 +1,337 @@
 'use strict';
 
-// write code here
+const tableHead = document.querySelector('thead');
+const allRows = [...document.querySelector('table').rows];
+const minAge = 18;
+const maxAge = 90;
+
+const sortingColumns = (e) => {
+  const updatedAllRows = [...document.querySelector('table').rows];
+  const sortByColumn = e.target.innerText;
+  const tableDataToSort = [];
+
+  for (let i = 1; i < updatedAllRows.length - 1; i++) {
+    const cells = updatedAllRows[i].cells;
+    const person = {};
+
+    person.Name = cells[0].innerText;
+    person.Position = cells[1].innerText;
+    person.Office = cells[2].innerText;
+    person.Age = cells[3].innerText;
+    person.Salary = cells[4].innerText;
+
+    tableDataToSort.push(person);
+  }
+
+  if (e.target.dataset.clicked === 'true') {
+    if (sortByColumn === 'Salary') {
+      tableDataToSort.sort((a, b) => {
+        const toNumberA = a.Salary.match(/\d+/g).join('');
+        const toNumberB = b.Salary.match(/\d+/g).join('');
+
+        return toNumberB - toNumberA;
+      });
+    } else {
+      tableDataToSort.sort((a, b) =>
+        b[sortByColumn].localeCompare(a[sortByColumn]));
+    }
+    e.target.dataset.clicked = false;
+  } else {
+    const clicedTitile = document.querySelector('[data-clicked="true"]');
+
+    if (clicedTitile) {
+      clicedTitile.dataset.clicked = false;
+    }
+
+    if (sortByColumn === 'Salary') {
+      tableDataToSort.sort((a, b) => {
+        const toNumberA = a.Salary.match(/\d+/g).join('');
+        const toNumberB = b.Salary.match(/\d+/g).join('');
+
+        return toNumberA - toNumberB;
+      });
+    } else {
+      tableDataToSort.sort((a, b) =>
+        a[sortByColumn].localeCompare(b[sortByColumn]));
+    }
```

Comment:
looks like duplication. I can suggest another approach. split callback choosing and order choosing logic. Use variable for a callback, set it based on clicked column, and then set the order flag, and in the end use them for sorting. Fell free to ask if you have questions